### PR TITLE
Replace token data with subgraph calls

### PR DESCRIPTION
**Pull request by neokry**
## Description

- replaces zora api calls with subgraph calls for token data

## Motivation & context

- reduce reliance on zora apis
- prepare for L2 deployments
- prepare for new protocol changes

## Code review

- does token data match with zora api data?
- do tokens on explore, detail, proposals and profile page properly display

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have done a self-review of my own code
- [x] Any new and existing tests pass locally with my changes
- [x] My changes generate no new warnings (lint warnings, console warnings, etc)

### CODE REVIEW

File: apps/web/src/pages/profile/[user].tsx [hunk 1]
```diff
@@ -168,20 +168,31 @@ const ProfilePage: NextPageWithLayout<ProfileProps> = ({ userAddress }) => {
...
-                {tokens?.tokens.map((x, i) => (
-                  <Link
-                    key={i}
-                    href={`/dao/${x.tokenContract?.collectionAddress}/${x.tokenId}`}
-                  >
-                    <TokenPreview name={x.name} image={x.image?.url} />
-                  </Link>
-                ))}
-              </Grid>
+              {(tokens?.tokens.length || 0) > 0 ? (
```
Comment (taayyohh) [inline apps/web/src/pages/profile/[user].tsx:171]: can this be `!!tokens?.tokens.length` i think that `0` is falsy and so `!!0 === true`

File: apps/web/src/pages/profile/[user].tsx [hunk 2]
```diff
@@ -168,20 +168,31 @@ const ProfilePage: NextPageWithLayout<ProfileProps> = ({ userAddress }) => {
...
+                >
+                  <Text color="text3">No more DAO tokens found.</Text>
+                </Flex>
+              )}
+
               <Pagination
                 onNext={handlePageForward}
                 onPrev={handlePageBack}
-                isLast={!tokens?.pageInfo?.hasNextPage}
+                isLast={(tokens?.tokens.length || 0) < 1}
```
Comment (taayyohh) [inline apps/web/src/pages/profile/[user].tsx:195]: same thing here `!tokens?.tokens.length` should do the trick

**Comment by neokry** [comment]
thanks for the review @taayyohh just updated these

File: apps/web/src/data/subgraph/requests/tokenQuery.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    return undefined
+  }
+
+  return {
+    id: token.tokenId,
+    ...omitBy(
+      {
+        owner: token.owner || undefined,
+        name: token.name || undefined,
+        description: token.dao.description || undefined,
```
Comment (nadiakiamilev) [inline apps/web/src/data/subgraph/requests/tokenQuery.ts:41]: do we need the undefined fallback assignment if these values are already possibly undefined?

File: apps/web/src/modules/create-proposal/components/UpgradeCard.tsx [hunk 4]
```diff
@@ -82,7 +61,7 @@ export const UpgradeCard = ({
             borderRadius={'small'}
             className={image}
           >
-            {!!imgurl && !error && (
+            {!!imgurl && (
```
Comment (nadiakiamilev) [inline apps/web/src/modules/create-proposal/components/UpgradeCard.tsx:64]: same comment as in the other pr!

**Pull request merged**